src/components/FormType.tsx

```diff
@@ -0,0 +1,123 @@
+import { useContext } from "react"
+import { ModalContext } from "~/state/ModalContext"
+
+type ModalStateSetter = (args_0: boolean) => void
+
+interface GitHubFormTypeProps {
+  modals: [ModalStateSetter, ModalStateSetter];
```

Comment:
Why a tuple type here instead of something that's more descriptive about what state you're setting e.g. an object type?

Reply:
No reason here. I am not worried on how the type is written as of now. I'm certain these files would change since my goal is to make it work e2e. I do agree that this type should be improved in terms of how it is written.

Reply:
Right but even in the short term this is not the best design. You have a tuple where both types are the same type and can easily be mixed up just be swapping their places. Also things are unnnamed on top of that. It's barely more effort to have something like

```
{ [key: string]: ModalStateSetter }
```

or

```
{
  myXModal: ModalStateSetter,
  myYModal: ModalStateSetter
}
```

Reply:
Ohhhh I see what you mean. Yeah this is something that needs to be changed.

Reply:
> You have a tuple where both types are the same type and can easily be mixed up just be swapping their places

This type of problem does happen in big codebases and it's really really nasty to debug